pkg/config/virtio.go

```diff
@@ -691,15 +777,26 @@ func (dev *USBMassStorage) SetReadOnly(readOnly bool) {
 // StorageConfig configures a disk device.
 type StorageConfig struct {
 	DevName   string `json:"devName"`
-	ImagePath string `json:"imagePath"`
+	ImagePath string `json:"imagePath,omitempty"`
+	URI       string `json:"uri,omitempty"`
 	ReadOnly  bool   `json:"readOnly,omitempty"`
 }
 
 func (config *StorageConfig) ToCmdLine() ([]string, error) {
-	if config.ImagePath == "" {
-		return nil, fmt.Errorf("%s devices need the path to a disk image", config.DevName)
+	if config.ImagePath != "" && config.URI != "" {
+		return nil, fmt.Errorf("%s devices cannot have both path to a disk image and a uri to a remote block device", config.DevName)
+	}
+	if config.ImagePath == "" && config.URI == "" {
+		return nil, fmt.Errorf("%s devices need a path to a disk image or a uri to a remote block device", config.DevName)
+	}
```

Comment:
Only a comment (or just me thinking out loud), I'm not asking for changes in response to this comment.
We could having both `ImagePath` and `URI` in `StorageConfig`, and avoid these checks with something like:

```
type StorageConfig struct {
	DevName   string `json:"devName"`
	ReadOnly  bool   `json:"readOnly,omitempty"`
}

type DiskStorageConfig {
    StorageConfig
    ImagePath string `json:"imagePath,omitempty"`
}

type NetworkBlockDevice {
    StorageConfig
    URI       string `json:"uri,omitempty"`
}
```

If it works, this makes sure we'll never get a NetworkBlockDevice with ImagePath set as it does not have this member at all.
This adds one more almost empty type though, and may be breaking unserializing json on upgrades (?).

Reply:
I like this idea but I propose to work on it in a follow-up PR. I see that StorageConfig is linked to many other things like NVMExpressController, VirtioBlk... I think it's better to have this refactoring separated for an easier review/testing

Reply:
I'll open an issue if you agree

Reply:
Yes, a follow-up issue is good, as I said:
> Only a comment (or just me thinking out loud)